src/openzaak/notifications/viewsets.py

```diff
@@ -36,3 +43,43 @@ def _message(self, data, instance=None):
                 )
 
                 transaction.on_commit(lambda msg=message: send_notification.delay(msg))
+
+
+type CloudEvenHandler = Callable[[CloudEvent], None | NoReturn]
+
+
+class CloudEventWebhook(APIView):
+    """Webhook that handles incoming CloudEvents via POST"""
+
+    handlers: set[CloudEvenHandler] = set()
+
+    @classmethod
+    def register_handler[T: CloudEvenHandler](cls, f: T, /) -> T:
+        cls.handlers.add(f)
+        return f
+
+    def post(self, request: Request):
+        try:
+            event: CloudEvent = from_http(request.headers, request.body)
+        except GenericException as e:
+            raise ValidationError(
+                {"cloudevent": [str(e)]}, code="malformed-cloudevent"
```

Comment:
here it's also maybe better to log the error and not return the full traceback as part of the API response?

Reply:
Ah you're right, sorry, I also didn't realize this is just the exception raised by `cloudevents` while parsing the event. Let's revert back `str(e)`, as this does give a clearer message:

with `str(e)`:

```
{
      "name": "cloudevent",
      "code": "malformed-cloudevent",
      "reason": "Found invalid specversion 1.a0"
}
```

with `__class__.__name__`:

```
{
      "name": "cloudevent",
      "code": "malformed-cloudevent",
      "reason": "InvalidRequiredFields"
}
```